### PR TITLE
Fix Reed Solomon decoder

### LINKED ISSUE #36: RS Decoder isn't correcting for bit errors
After finally getting some time to more thoroughly test the `decoder` it appears that the Reed Solomon portion of the process is not actually correcting for any bit errors. I was suspicious of this when I handed off the FEC but I wasn't able to validate it until now. This is causing all the CRC checks to fail which causes the decoder to abort since it isn't able to find any valid OTI header. 

Here is a hexdump of the before and after we Reed Solomon decode a packet whose symbol data was all zeroes: 

fec-error
 
The first 16 bytes highlighted in blue is the OTI header, which is actually in perfect condition. The bytes highlighted in purple are those with bit errors, there are 7 errors which RS should be able to correct for. The 7 bytes highlighted in green are the parity bits that we attached when we RS encoded the original symbol. 

As you can see after we RS decode the packet and consume the parity bits, the errors are still present. 

Unsure how to proceed, my guess is that this an issue of not using the RSCode library correctly. The library was sparsely documented and only had a minimal example. So, we may be missing some sort of initialization or parameters for the decoding routine.

### PR DESCRIPTION
**Pull request by digital-nomad8620**
PR fixes #36 and addresses a memory leak in the OpenFEC library as well as clean up some code. 

The reason why the RS decoder wasn't doing anything was because we did not initialize the library. (Thanks @Cam57  for spotting that). However, after initializing the library, the program kept segfaulting because I had used the `PROT_READ` flag when I mmapped the received file into memory. Changing this `PROT_WRITE` fixes the issue, but as a side-effect the file on disk will be modified during the decoding process to correct for any bit errors. This is probably fine since the file being modified is an intermediate file anyways. 

Fixing these issues appeared to solve the problem, but all the CRC checks kept failing which caused the Decoder to abort. Further investigation yielded that we were correcting for *some* but not all the bit errors.  After re-reading the Barcelona paper I realized that our `RSCODE_NPAR` value of 7 was way too low. 

<blockquote>
For deep space communications there is the Reed-Solomon standard (255,223) proposed by the CCSDS in which there are 223 symbols of information and 32 symbols of parity in a 255 encoded packet. The packet length of this standard is 255 symbols because it works over a Galois field of dimension 2^8 and the 32 symbols of parity permit the detection and correction of any 16 erroneous symbols in the packet. 
</blockquote>

The key thing to note here is that we can **only** correct for 16 bad symbols in any given RS code block. Since we have 5 blocks for each LDPC frame that `tx` transmits we can only correct for at **most** 80 errors before the CRC check will fail causing the decoder to assume the RS decoding did not work. I have some thoughts on how to make the decoder more robust but will detail that in our next meeting and make a separate issue for it. 

After adjusting the the `RSCODE_NPAR` value to match the CCSDS standard of (255,223) I was able to successfully encode, transmit, receive, and decode a test file with a bit error rate of ~0.7%.  I was also able to finally get preliminary results from the `sweep.py` program However I ran into another smaller issue which I recorded in #40

### CODE REVIEW

**Comment by digital-nomad8620** [comment]
@CD-W You're going to want to rebase off of main when this branch is merged. It fixes the memory leak issues from OpenFEC. And also, fixes the decoder.

**Review by ghost — APPROVED** [review]
Approved.

**Review by khanu263 — APPROVED** [review]
Looks good to me as well. Out of curiosity, how much do these CCSDS RS parameters increase data overhead?

**Comment by digital-nomad8620** [comment]
> Looks good to me as well. Out of curiosity, how much do these CCSDS RS parameters increase data overhead?

@khanu263 Let's see. We went from being able to encode 1224 bytes of data per packet to 1099 bytes per packet. So there's an extra 125 bytes of overhead per packet if I did my math right.

File: libdxwifi/fec.c [hunk 1]
```diff
@@ -206,11 +209,15 @@ ssize_t dxwifi_decode(void* encoded_msg, size_t msglen, void** out) {
...
-        for(size_t i = 0; i < DXWIFI_RSCODE_BLOCKS_PER_FRAME; ++i) {
-            void* message  = offset(ldpc_frame, i, RSCODE_MAX_MSG_LEN);
-            void* codeword = offset(rs_ldpc_frame, i, RSCODE_MAX_LEN);
+        for(size_t j = 0; j < DXWIFI_RSCODE_BLOCKS_PER_FRAME; ++j) {
+            void* message  = offset(ldpc_frame, j, RSCODE_MAX_MSG_LEN);
+            void* codeword = &rs_ldpc_frame->blocks[j];
 
             decode_data(codeword, RSCODE_MAX_LEN);
+
+            //log_hexdump(message, RSCODE_MAX_MSG_LEN);
```
Comment (digital-nomad8620) [inline libdxwifi/fec.c]: Remove commented out code.

**Review by digital-nomad8620 — COMMENTED** [review]
Need to remove commented out code then I'll be done with this PR

**Pull request merged**